src/libLookup/Lookup.cpp

```diff
@@ -5436,145 +5441,196 @@ void Lookup::RectifyTxnShardMap(const uint32_t oldNumShards,
-      continue;
-    }
-    vector<Peer> toSend;
-    if (i < numShards) {
-      {
-        lock_guard<mutex> g(m_mediator.m_ds->m_mutexShards);
-        uint16_t lastBlockHash = DataConversion::charArrTo16Bits(
-            m_mediator.m_txBlockChain.GetLastBlock().GetBlockHash().asBytes());
-        uint32_t leader_id = m_mediator.m_node->CalculateShardLeaderFromShard(
-            lastBlockHash, m_mediator.m_ds->m_shards.at(i).size(),
-            m_mediator.m_ds->m_shards.at(i));
-        LOG_EPOCH(INFO, m_mediator.m_currentEpochNum,
-                  "Shard leader id " << leader_id);
-
-        auto it = m_mediator.m_ds->m_shards.at(i).begin();
-        // Lookup sends to NUM_NODES_TO_SEND_LOOKUP + Leader
-        unsigned int num_node_to_send = NUM_NODES_TO_SEND_LOOKUP;
-        for (unsigned int j = 0; j < num_node_to_send &&
-                                 it != m_mediator.m_ds->m_shards.at(i).end();
-             j++, it++) {
-          if (distance(m_mediator.m_ds->m_shards.at(i).begin(), it) ==
-              leader_id) {
-            num_node_to_send++;
-          } else {
-            toSend.push_back(std::get<SHARD_NODE_PEER>(*it));
-            LOG_GENERAL(INFO, "Sent to node " << get<SHARD_NODE_PEER>(*it));
-          }
+      for (auto const& i : *m_mediator.m_DSCommittee) {
+        if (toSend.size() < NUM_NODES_TO_SEND_LOOKUP &&
+            i.second != dsLeader.second) {
+          toSend.push_back(i.second);
         }
-        if (m_mediator.m_ds->m_shards.at(i).empty()) {
-          continue;
+
+        if (toSend.size() >= NUM_NODES_TO_SEND_LOOKUP) {
+          break;
         }
       }
+    }
 
-      P2PComm::GetInstance().SendBroadcastMessage(toSend, msg);
+    P2PComm::GetInstance().SendBroadcastMessage(toSend, msg);
 
-      DeleteTxnShardMap(i);
-    } else if (i == numShards) {
-      // To send DS
-      {
-        lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
+    LOG_GENERAL(INFO, "[DSMB]"
+                          << " Sent DS the txns");
 
-        if (m_mediator.m_DSCommittee->empty()) {
-          continue;
-        }
+    DeleteTxnShardMap(shardId);
+  }
+}
 
-        // Send to NUM_NODES_TO_SEND_LOOKUP which including DS leader
-        PairOfNode dsLeader;
-        if (Node::GetDSLeader(m_mediator.m_blocklinkchain.GetLatestBlockLink(),
-                              m_mediator.m_dsBlockChain.GetLastBlock(),
-                              *m_mediator.m_DSCommittee, dsLeader)) {
-          toSend.push_back(dsLeader.second);
-        }
+void Lookup::SendTxnPacketToDS(const uint32_t oldNumShards,
+                               const uint32_t newNumShards) {
+  LOG_MARKER();
 
-        for (auto const& i : *m_mediator.m_DSCommittee) {
-          if (toSend.size() < NUM_NODES_TO_SEND_LOOKUP &&
-              i.second != dsLeader.second) {
-            toSend.push_back(i.second);
-          }
+  SendTxnPacketPrepare(oldNumShards, newNumShards);
```

Comment:
This will generates txns for all shards ( not just ds shard ). This one most likely will cause high nonce issue on ds shard because txn pkt to ds will be sent immediately.